private/model/api/operation.go

```diff
@@ -44,6 +44,25 @@ func (o *Operation) HasOutput() bool {
 	return o.OutputRef.ShapeName != ""
 }
 
+func (o *Operation) GetSigner() string {
+	if o.AuthType == "v4-unsigned-body" {
+		o.API.imports["github.com/aws/aws-sdk-go/aws/signer/v4"] = true
+	}
+
+	buf := bytes.NewBuffer(nil)
+
+	switch o.AuthType {
+	case "none":
+		buf.WriteString("req.Config.Credentials = credentials.AnonymousCredentials")
+	case "v4-unsigned-body":
+		buf.WriteString("req.Handlers.Sign.Clear()\n")
```

Comment:
Custom request handlers that are added at the service or session level would of been applied to the request when `c.newRequest(` is called.  A `Clear()` here would remove any  of those custom request handlers added to the `Sign` list.  Only custom request handlers added after the request is created would be preserved. This was the reasoning for explicitly removing the known signers. 

Though fun to consider if this trait were to be applied to a service like `S3` anyone using a custom v2 signer that the SDK has no knowledge of would be removed on a `Clear()`, or incorrectly left in on a remove by name. I'm not sure the best way to handle this. Situations like this is where the concept of the request handlers being a list breaks down, and a singular function call is needed.

I'm not sure `Clear()` can be used without additional logic. Partly because custom handlers added by the user, and handlers like [`BuildContentLengthHandler`](https://github.com/aws/aws-sdk-go/blob/master/aws/defaults/defaults.go#L68-L82) are added to the `Sign` list.